sqld/src/http/auth.rs

```diff
@@ -52,3 +77,37 @@ impl Authorizer for BasicAuthAuthorizer {
         }
     }
 }
+
+/// Bearer token authentication authorizer.
+pub struct BearerAuthAuthorizer {
+    decoding_key: DecodingKey,
+}
+
+impl BearerAuthAuthorizer {
+    fn validate_token(&self, token: &str) -> Result<bool> {
+        // Once we start verifying claims, token will become useful
+        let _token = jsonwebtoken::decode::<JsonValue>(
+            token,
+            &self.decoding_key,
+            &Validation::new(jsonwebtoken::Algorithm::RS256),
+        )?;
+
+        Ok(true)
+    }
+}
+
+impl Authorizer for BearerAuthAuthorizer {
+    fn is_authorized(&self, req: &Request<Body>) -> bool {
+        let headers = req.headers();
+        let actual_auth = headers.get(hyper::header::AUTHORIZATION);
+        if let Some(Ok(actual_auth)) = actual_auth.map(|a| a.to_str()) {
+            if !actual_auth.starts_with("Bearer ") {
+                return false;
+            }
+            let token = &actual_auth[7..];
```

Comment:
I would split white space here, no?

Reply:
I don't understand the question - is it about correctness? The format is defined [here](https://datatracker.ietf.org/doc/html/rfc6750#section-2.1) specifically as "Bearer <1SP> <token>", so the token is always exactly at at offset 7, because the standard requires 1 space. And I hope it's easy enough to reason about the origin of "7", given that the string is 2 lines above.

Reply:
NVM, I didn't know the spec was explicit about it being exactly one space, that works 👍

Reply:
me neither, I just hoped it's the case and frantically went over the RFCs to prove my point :innocent: got lucky on that one